pkgs/by-name/xa/xar/package.nix

```diff
@@ -0,0 +1,175 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  applyPatches,
+
+  autoreconfHook,
+
+  # Required dependencies.
+  zlib,
+  libxml2,
+
+  # Optional dependencies.
+  withExt2 ? lib.meta.availableOn stdenv.hostPlatform e2fsprogs,
+  e2fsprogs,
+  withAcl ? lib.meta.availableOn stdenv.hostPlatform acl,
+  acl,
+  withBzip2 ? true,
+  bzip2,
+  withLzma ? true,
+  xz, # lzma
+
+  # Platform-specific dependencies.
+  musl-fts,
+  openssl,
+  darwin, # CommonCrypto
+
+  gitUpdater,
+
+  # for tests
+  xar,
+  testers,
+  python3,
+  libxslt, # xsltproc
+  runCommand,
+  makeWrapper,
+  buildPackages,
+}:
+let
+  pname = "xar";
+  version = "498";
+
+  pythonForTests = py: py.withPackages (p: [ p.xattr ]);
+
+  src = applyPatches {
+    name = "source";
+    src = fetchurl {
+      url = "https://opensource.apple.com/tarballs/xar/xar-${version}.tar.gz";
+      hash = "sha256-nO5PgLls9ZLMxUWk/dUeTaSlvTtHNJAWN9Z7BD7/PHU=";
+    };
+    patches = lib.filesystem.listFilesRecursive ./patches;
+  };
```

Comment:
I guess this can’t use `fetchFromGitHub` for bootstrap reasons? It might be nice to use GitHub’s tarball URL, though, for more of a guarantee that it matches the Git repository. GitHub seems to be Apple’s main FOSS distribution method at this point anyway.

Reply:
`fetchFromGitHub` is allowed in the Darwin bootstrap because it’s really a `fetchzip`. It’s obly `fetchgit` that’s a problem because it depends on git, which is not in the bootstrap tools.

Reply:
Right. Then I think it’d be nice to use `fetchFromGitHub` here.

Reply:
Updated to use `fetchFromGitHub`.